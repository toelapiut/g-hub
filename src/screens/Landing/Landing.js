import Search from '../../components/Search';
import Image from 'next/image';
import styles from './landing.module.css';
import Head from 'next/head';
import React,{Fragment} from 'react';


export const Landing = () => {
  return (
    <Fragment>
      <Head>
        <title>Welcome Ghub Clone - safe space to find coders</title>
      </Head>
      <div className={styles.contain}>
        <div className={styles.column}>
          <h1 className={styles.title}>Where the world builds software</h1>
          <h4 className={styles.subtitle}>Millions of developers and companies build, ship, and maintain their software on
            GitHubâ€”the largest and most
            advanced development platform in the world.</h4>
          <Search/>
        </div>
        <div className={styles.column}>
          <Image
            src="/images/global.png"
            alt="global image"
            width={600}
            height={600}
          />
        </div>
      </div>
    </Fragment>
  );
};