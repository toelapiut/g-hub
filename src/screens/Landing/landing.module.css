.contain {
    @apply flex flex-row;
    min-height: calc(100vh - 5rem);
    overflow-x: scroll;
}

.column {
    max-width: 40rem;
    @apply flex flex-col justify-center
}

.title, .subtitle {
    font-size: 3rem;
    line-height: 1.2;
    @apply font-sans font-extrabold;

}

.subtitle {
    @apply text-xl font-light pt-3 mb-10;
}