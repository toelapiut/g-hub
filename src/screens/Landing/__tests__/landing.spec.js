import React from 'react';
import Landing from '../';
import renderer from 'react-test-renderer';
import {describe, it} from '@jest/globals';

describe('landing page Component', () => {

  it('matches the landing page snapshots', () => {
    const tree = renderer.create(
      <Landing/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});