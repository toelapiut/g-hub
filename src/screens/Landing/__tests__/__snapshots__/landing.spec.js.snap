// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page Component matches the landing page snapshots 1`] = `
<div
  className="contain"
>
  <div
    className="column"
  >
    <h1
      className="title"
    >
      Where the world builds software
    </h1>
    <h4
      className="subtitle"
    >
      Millions of developers and companies build, ship, and maintain their software on GitHubâ€”the largest and most advanced development platform in the world.
    </h4>
    <div>
      <div
        className="contain"
        style={Object {}}
      >
        <p
          className="label"
          style={Object {}}
        >
          
           
        </p>
        <div
          className="input"
        >
          <span
            className="sideText"
            style={Object {}}
          />
          <input
            className="inputField"
            name="Search"
            onChange={[Function]}
            onFocus={[Function]}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="column"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjAwIiBoZWlnaHQ9IjYwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="global image"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
